src/ObjLoading/Game/T6/BSP/LoaderBSP_T6.cpp

```diff
@@ -0,0 +1,60 @@
+#include "LoaderBSP_T6.h"
+
+#include "BSPCreator.h"
+#include "BSPUtil.h"
+#include "Linker/BSPLinker.h"
+
+namespace
+{
+    using namespace BSP;
+
+    class BSPLoader final : public IAssetCreator
+    {
+    public:
+        BSPLoader(MemoryManager& memory, ISearchPath& searchPath, Zone& zone)
+            : m_memory(memory),
+              m_search_path(searchPath),
+              m_zone(zone)
+        {
+        }
+
+        std::optional<asset_type_t> GetHandlingAssetType() const override
```

Comment:
nit: Not _that_ important but can be helpful sometimes: There is a c++ feature called [`nodiscard`](https://en.cppreference.com/w/cpp/language/attributes/nodiscard) that raises a compiler warning if the return value of a function is not used.

Helpful at functions that do nothing other than return a value.
Generally mostly applicable when a method is static or const.

adjusted in: 8f08aca15a2c13438e26cfe25e8aa17087ee6e3a